snapcraft/plugins/ruby.py

```diff
@@ -0,0 +1,193 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2017 James Beedy <jamesbeedy@gmail.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""The ruby plugin is useful for ruby based parts.
+The plugin uses gem to install dependencies from a `Gemfile`.
+This plugin uses the common plugin keywords as well as those for "sources".
+For more information check the 'plugins' topic for the former and the
+'sources' topic for the latter.
+Additionally, this plugin uses the following plugin-specific keywords:
+    - gems:
+      (list)
+      A list of gems to install.
+    - use-bundler
+      (boolean)
+      Use bundler to install gems from a Gemfile (defaults 'false').
+    - ruby-version:
+      (string)
+      The version of ruby you want this snap to run.
+"""
+
+import glob
+import logging
+import os
+import platform
+import re
+
+from snapcraft import BasePlugin, file_utils
+from snapcraft.internal.errors import SnapcraftEnvironmentError
+from snapcraft.sources import Tar
+
+
+logger = logging.getLogger(__name__)
+
+
+class RubyPlugin(BasePlugin):
+
+    @classmethod
+    def schema(cls):
+        schema = super().schema()
+
+        schema['properties']['use-bundler'] = {
+            'type': 'boolean',
+            'default': False
+        }
+        schema['properties']['ruby-version'] = {
+            'type': 'string',
+            'default': '2.4.0'
+        }
+        schema['properties']['gems'] = {
```

Comment:
Picking up from the previous PR:
Bundler has a `--without` switch - how about `gems: [spam, -eggs]`, akin to the syntax used with `prime`, to support that?
@kyle suggested the more explicit `bundler-without` for the same feature.

Reply:
Not a requirement for this PR, we can grow it. Although as I mentioned in the other PR, this makes no sense for individual gems.

Reply:
Yeah, just wanted to make sure the discussion wasn't lost. Maybe it's worth taking to the forum anyway

Regarding the builder `--without` feature: I'm guessing my proposal is too confusing then. I didn't mean to suggesting anything other than what builder supports, which is excluding gems by group name. So, nevermind.